client/post-editor/editor-confirmation-sidebar/index.jsx

```diff
@@ -3,54 +3,113 @@
 
 /**
  * Internal dependencies
  */
 import RootChild from 'components/root-child';
 import Button from 'components/button';
+import EditorVisibility from 'post-editor/editor-visibility';
+import { editPost } from 'state/posts/actions';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getEditorPostId } from 'state/ui/editor/selectors';
+import { getEditedPost } from 'state/posts/selectors';
 
 class EditorConfirmationSidebar extends React.Component {
 	static propTypes = {
-		hideSidebar: React.PropTypes.func,
-		isActive: React.PropTypes.bool,
+		closeOverlay: React.PropTypes.func,
+		closeSidebar: React.PropTypes.func,
+		isOverlayActive: React.PropTypes.bool,
+		isSidebarActive: React.PropTypes.bool,
 		onPublish: React.PropTypes.func,
+		post: React.PropTypes.object,
+		site: React.PropTypes.object,
+		savedPost: React.PropTypes.object,
+		onPrivatePublish: React.PropTypes.func,
+	};
+
+	closeOverlayAndSidebar = () => {
+		this.props.closeSidebar();
+		this.props.closeOverlay();
 	};
 
 	closeAndPublish = () => {
-		this.props.hideSidebar();
+		this.closeOverlayAndSidebar();
 		this.props.onPublish( true );
 	};
 
+	renderPrivacyControl() {
+		if ( ! this.props.post ) {
+			return;
+		}
+
+		const { password, type, status } = this.props.post || {};
+		const isPrivateSite = this.props.site && this.props.site.is_private;
+		const savedStatus = this.props.savedPost ? this.props.savedPost.status : null;
+		const savedPassword = this.props.savedPost ? this.props.savedPost.password : null;
+		const props = {
+			onPrivatePublish: this.props.onPrivatePublish,
+			isPrivateSite,
+			type,
+			password,
+			status,
+			savedStatus,
+			savedPassword
+		};
+
+		return (
+			<EditorVisibility { ...props } />
+		);
+	}
+
 	render() {
 		return (
 			<RootChild>
 				<div className={ classnames( {
 					'editor-confirmation-sidebar': true,
-					'is-active': this.props.isActive,
+					'is-active': this.props.isOverlayActive || this.props.isSidebarActive,
 				} ) } >
 					<div className={ classnames( {
 						'editor-confirmation-sidebar__overlay': true,
-						'is-active': this.props.isActive,
-					} ) } onClick={ this.props.hideSidebar }></div>
+						'is-active': this.props.isOverlayActive,
+					} ) } onClick={ this.closeOverlayAndSidebar } />
 					<div className={ classnames( {
 						'editor-confirmation-sidebar__sidebar': true,
-						'is-active': this.props.isActive,
+						'is-active': this.props.isSidebarActive,
 					} ) }>
 						<div className="editor-confirmation-sidebar__ground-control">
-							<div className="editor-confirmation-sidebar__cancel" onClick={ this.props.hideSidebar }>
+							<div className="editor-confirmation-sidebar__cancel" onClick={ this.closeOverlayAndSidebar }>
 								{ this.props.translate( 'Cancel' ) }
 							</div>
 							<div className="editor-confirmation-sidebar__action">
 								<Button onClick={ this.closeAndPublish } compact>{ this.props.translate( 'Publish' ) }</Button>
```

Comment:
The underlying publish button is compact. I believe this just appears smaller because we removed the text-transform. Can you please surface the discussion of this / changes involved outside of this PR as the button wasn't touched in this context?